### PR TITLE
List structure for To-do List

### PR DESCRIPTION
**Pull request by dhirajsingha143**
### TO-DO LIST Milestone-1
- Setup web pack
- Installed HTML plugin for webpack
- Installed CSS plugin for webpack
- Installed webpack dev server for local hosting
- Created HTML, CSS and JS files
- Add basic markup structure and styling
- Add functionality and used ES6 syntax
- Installed Linters actions

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import './style.css';
+
+const updateLocalStorage = (toDoListArray) => {
+  localStorage.setItem('toDoList', JSON.stringify(toDoListArray));
+};
```
Comment (TadesseDev) [inline src/index.js]: - [ ] I'm impressed with the step you have taken @dhirajsingha143, But as a main requirement for this project the project should have an initial array with some simple *to do tasks* (array of objects) with each object containing three keys - description \[`string`\], completed \[`bool`\], index: \[`number`\].

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  height: 90%;
+}
+
+.toDo_list {
+  min-width: 300px;
+  width: 50%;
+  max-width: 1000px;
+  box-shadow: 10px 14px 14px -5px var(--black);
+  border: 1px solid var(--black);
+}
```
Comment (TadesseDev) [inline src/style.css:45]: - [ ] Dear @dhirajsingha143, Your styling is great, But we need to stick to the given design. You don't have to do all the functionalities as in the provided vide but the layout should lookalike the one given.

### Your design 
Screenshot 2023-01-20 203235

### Given design
Screenshot 2023-01-20 203300

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @dhirajsingha143 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7

## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitHub Flow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    editTask(e, toDoTasks);
+  }
+});
+
+todoList.addEventListener('click', (e) => {
+  if (e.target.closest('.todo-list-li-cross')) {
+    deleteTask(e, toDoTasks);
+  }
+});
+
```
Comment (jerryowusu) [inline src/index.js]: As requested by the previous reviewer, you do not need to implement all the tasks at once, kindly focus on the current activity as shown below :point_down: As a main requirement at this stage of  the project you just have to have an initial array with some simple to do tasks (array of objects) with each object containing three keys - description [string], completed [bool], index: [number]. 


Screenshot from 2023-01-21 06-18-53

File: src/style.css [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+.container {
+  display: flex;
+  padding: 14px;
+}
+
+.task-input {
+  /* stylelint-disable-next-line */
```
Comment (jerryowusu) [inline src/style.css]: [OPTIONAL] The linters are suppose to guide you to achieve best standard practices, so it is not a good practice to disable parts of it. Kindly consider removing this line :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @dhirajsingha143  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

no-we-were-so-close-pauline-turner-brooks

 ## Highlights
Nice commit messages :heavy_check_mark: 
Used github flow :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi @dhirajsingha143 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- All requirements met for this milestone.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**